uncover/agent/hunterhow/hunterhow.go

```diff
@@ -0,0 +1,120 @@
+package hunterhow
+
+import (
+	"encoding/json"
+	"errors"
+	"net/http"
+
+	"github.com/projectdiscovery/uncover/uncover"
+)
+
+const (
+	baseURL      = "https://api.hunter.how/"
+	baseEndpoint = "search"
+	Size         = 100
+)
+
+type Agent struct {
+	options *uncover.AgentOptions
+}
+
+func New() (uncover.Agent, error) {
+	return &Agent{}, nil
+}
+
+func NewWithOptions(options *uncover.AgentOptions) (uncover.Agent, error) {
+	return &Agent{options: options}, nil
+}
+
+func (agent *Agent) Name() string {
+	return "hunterhow"
+}
+
+func (agent *Agent) Query(session *uncover.Session, query *uncover.Query) (chan uncover.Result, error) {
+	if session.Keys.HunterHowToken == "" {
+		return nil, errors.New("empty hunterhow keys")
+	}
+
+	results := make(chan uncover.Result)
+
+	go func() {
+		defer close(results)
+
+		numberOfResults := 0
+
+		pageQuery := 1
+
+		for {
+			hunterhowRequest := &Request{
+				Query:    query.Query,
+				PageSize: query.Limit,
+				Page:     pageQuery,
+			}
+
+			if numberOfResults > query.Limit {
+				break
+			}
+
+			hunterhowResponse := agent.query(hunterhowRequest.buildURL(session.Keys.HunterHowToken), session, results)
+			if hunterhowResponse == nil {
+				break
+			}
+
+			if len(hunterhowResponse) == 0 {
+				break
+			}
+
+			numberOfResults += len(hunterhowResponse)
+			pageQuery += 1
+		}
+	}()
+
```

Comment:
@xm1k3 ,not sure if you want to implement in this PR or a new one . but just to point out this implementation looks good if no of results < 1000 (10 requests are made in serial) but if there are more than 1000> requests it takes time which shouldn't be the case . 
From api docs  it returns ` "total":5994` in response so after first request we can get total results and calculate number of pages available and fetch them according to `limit` cli option in parallel, this way it won't take that much time 

```
so query returns 5994 results it is ~60 req 
60 req in serial (current implementation) takes ~40-50 sec 
60 req in parallel takes ~10 sec
*estimated time here is approx and depends on various factors
```